src/bus.rs

```diff
@@ -244,6 +247,40 @@ impl<B: UsbBus> UsbBusAllocator<B> {
             .expect("alloc_ep failed")
     }
 
+    /// Allocates an isochronous endpoint.
+    ///
+    /// # Arguments
+    ///
+    /// * `synchronization` - Type of synchronization used by the endpoint
+    /// * `usage` - Whether the endpoint is data, explicit feedback, or data+implicit feedback
+    /// * `payload_size` - Payload size in bytes.
+    /// * `interval` - Interval for polling, expressed in frames/microframes.  Must be 1 to 16,
+    ///    calculated as 2^(interval-1), see USB 2.0 section 9.6.6.
+    ///
+    /// # Panics
+    ///
+    /// Panics if endpoint allocation fails, because running out of endpoints or memory is not
+    /// feasibly recoverable.
+    #[inline]
+    pub fn isochronous<D: EndpointDirection>(
+        &self,
+        synchronization: IsochronousSynchronizationType,
+        usage: IsochronousUsageType,
+        payload_size: u16,
+        interval: u8,
+    ) -> Endpoint<'_, B, D> {
+        self.alloc(
+            None,
+            EndpointType::Isochronous {
+                synchronization,
+                usage,
+            },
+            payload_size,
+            interval,
+        )
+        .expect("alloc_ep failed")
```

Comment:
I would just pop the error out as a `Result<>` instead of unwrapping with the `expect()` call

Reply:
I'm in favour of this in principle, however the other allocator methods use `.expect()`  and I think it's probably better to either stick with that convention or change all of the allocators to return `Result<>`.  The second would be a breaking change, so might be better combined with other big-ish changes?

Reply:
I personally dislike deferring to current design, but I think you're right in this case that we don't gain anything via changing this expect (i.e. it does not remove format bloat).

We should spawn an issue to handle converting to `Result<>` return types to avoid format bloat

Reply:
I like the idea of splitting that issue out.  It might be a good improvement to combine with some more flexible memory allocation scheme; IIRC the current one (or maybe just be the ATSAMD implementation) just uses a fixed-size buffer, which as the comment says is a bit limiting.